kube-controllers/tests/fv/metrics_test.go

```diff
@@ -448,3 +565,91 @@ func getMetrics(metricsURL string) (string, error) {
 	}
 	return string(body), nil
 }
+
+func createNode(node string, k8sClient *kubernetes.Clientset) {
+	_, err := k8sClient.CoreV1().Nodes().Create(context.Background(),
+		&v1.Node{
+			TypeMeta:   metav1.TypeMeta{Kind: "Node", APIVersion: "v1"},
+			ObjectMeta: metav1.ObjectMeta{Name: node},
+			Spec:       v1.NodeSpec{},
+		},
+		metav1.CreateOptions{})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+}
+
+func createIPPool(name string, cidr string, calicoClient client.Interface) {
+	p := api.NewIPPool()
+	p.Name = name
+	p.Spec.CIDR = cidr
+	p.Spec.BlockSize = 26
+	p.Spec.NodeSelector = "all()"
+	p.Spec.Disabled = false
+	_, err := calicoClient.IPPools().Create(context.Background(), p, options.SetOptions{})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+}
+
+func allocatePodIpWithHandle(ip string, handle string, node string, pod string, calicoClient client.Interface) {
+	attrs := map[string]string{"node": node, "pod": pod, "namespace": "default"}
+	err := calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
+		IP: net.MustParseIP(ip), HandleID: &handle, Attrs: attrs, Hostname: node,
+	})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+}
+
+func allocateInterfaceIpWithHandle(ip string, handle string, node string, itype string, calicoClient client.Interface) {
+	attrs := map[string]string{"node": node, "type": itype}
+	err := calicoClient.IPAM().AssignIP(context.Background(), ipam.AssignIPArgs{
+		IP: net.MustParseIP(ip), HandleID: &handle, Attrs: attrs, Hostname: node,
+	})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+}
+
+func createPod(podName string, ip string, handle string, node string, k8sClient *kubernetes.Clientset, calicoClient client.Interface) {
+	pod := &v1.Pod{
+		TypeMeta:   metav1.TypeMeta{Kind: "Pod", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{Name: podName, Namespace: "default"},
+		Spec: v1.PodSpec{
+			NodeName: node,
+			Containers: []v1.Container{
+				{
+					Name:    "container1",
+					Image:   "busybox",
+					Command: []string{"sleep", "3600"},
+				},
+			},
+		},
+	}
+	pod, err := k8sClient.CoreV1().Pods("default").Create(context.Background(), pod, metav1.CreateOptions{})
+	ExpectWithOffset(1, err).NotTo(HaveOccurred())
+	pod.Status.PodIP = ip
+	pod.Status.Phase = v1.PodRunning
+	_, err = k8sClient.CoreV1().Pods("default").UpdateStatus(context.Background(), pod, metav1.UpdateOptions{})
```

Comment:
_Probably_ won't have any impact on the test, but to be slightly more realistic the ordering should go like this:

- Create pod in Pending state
- Allocate an IP in Calico IPAM
- Update the pod status to be Running, and to include the pod IP

This matches more closely the order that occurs on a real cluster, where the pod isn't marked running until an IP is allocated by the CNI plugin.